packages/api/src/middleware/auth.ts

```diff
@@ -73,6 +104,13 @@ function authFactory(params) {
     if ((params.admin && !isUIAdmin) || (params.anyAdmin && !user.admin)) {
       throw new ForbiddenError(`user does not have admin priviledges`);
     }
+    const accessRules = tokenObject?.access?.rules;
+    if (accessRules) {
+      const { httpPrefix } = req.config;
+      if (!isAuthorized(req.method, req.path, accessRules, httpPrefix)) {
```

Comment:
in `app.post("/hook/detection", ...` handler `req.path` is equal to `/hook/detection`
this means this whole system in the current state will not work at all.

Reply:
Good catch, thanks. I'll try to make this work using the `req.baseUrl` instead and add some tests to catch that scenario.

Reply:
Done!

Reply:
Rebased now so Github might have f'ed up the inceremental changes for you. The only commits I added since you last looked are these:
 - https://github.com/livepeer/livepeer-com/pull/483/commits/1e12ecb1cc6c1a90057d6e23132cec38ca293cb5
 - https://github.com/livepeer/livepeer-com/pull/483/commits/315ba1c99c73c32768825d981d14d8d6f155d591